src/ArgParse.jl

```diff
@@ -1937,10 +1937,10 @@ function preparse(c::Channel, state::ParserState, settings::ArgParseSettings)
             state.arg_delim_found = true
             state.token = nothing
             state.token_arg = nothing
-            shift!(args_list)
+            popfirst!(args_list)
             continue
         elseif startswith(arg, "--")
-            eq = search(arg, '=')
+            eq = findnext('=', arg)
             if eq != 0
```

Comment:
findnext returns nothing if the pattern is not found.

```
+            eq = findfirst(x -> isequal(x, '='), arg)
-            if eq != 0
+            if eq !== nothing
```

Reply:
I replaced `eq = findnext('=', arg)` with `eq = findfirst(equalto('='), arg)`.